### ISSUE TITLE
[improvement] Allow to jump to a page

### ISSUE THREAD

**Issue by asitemade4u** [sissbruecker/linkding#195]
- Currently Linkding only allows Previous, nearby page numbers, Next as choices
- Moreover, there are many cases where the user is brought back to the first page
- It would come very handy to be able to jump straight to a certain page number (for example the page one was at before he/she was brought back to the first page...)

**Comment by sissbruecker** [comment]
@asitemade4u 

> Moreover, there are many cases where the user is brought back to the first page

Can you elaborate on this? Adding a jump to page feature IMO is not a good solution to this specific problem (which is the only one I can find in your post). And there is actually logic in the application to ensure that you are returned to exactly the same page (query and page number) that you were on when:
- editing a bookmark
- archiving a bookmark
- deleting a bookmark
- bulk editing bookmarks

The only thing I can think of that is missing is adding a new bookmark. If there are any other cases where the application doesn't return you to your search after an action on the search results, then I would consider this a bug.

That you loose your search context when you click into some other part of the UI is expected. E.g. switching from search results to settings. Here you can always use the browsers backward navigation to get back to where you were.

Some additional thoughts on paging. I added the pagination with page numbers that linkding uses currently very reluctantly, previously you could only navigate one page back / forward. While opinionated, my reasoning was that if people have to crawl through pages of bookmarks, then something is obviously wrong with the UX, probably with the search function. If there are no additional use-cases for jumping to a page apart from the one you mentioned, my opinion here would be pretty much the same, no one should have to dig through specific pages of bookmarks in order to find something.

**Comment by asitemade4u** [comment]
I follow your thoughts on the matter.
My use case is that I imported (a lot of) links from different sources into Linkding and would now like to tag them.
Currently, there are not enough ways IMO to detect the ones that **have not been tagged**.
This soloution would be a nice midway between your desire for simplicity and my desire for efficiency:

1. add a way to search for "Not tagged entries"
2. refresh the view/selection when tags are added to entries
3. that way, one would always stay on 1st page and modify/tag the top of the list entries

**Comment by sissbruecker** [comment]
I don't understand what this new use case has to do with jumping to a page, can you elaborate on that? Even if you have to manually go through all pages ATM, then editing something on a page should still return you that page after saving the bookmark.

**Comment by asitemade4u** [comment]
You are right, I was digressing because I felt you were hell-bent on not doing it.
The fact is that sometimes due to refreshing, one is brought back to the first page. That happens in circumstances where you ared painstakingly tagging entries and it feels frustrating to:

1. remember on what page you were before being brought back to page 1
2. go back to the relevant page
3. scroll down the list of entries to treat the next one

As that isoften the case when tagging newly added (or imported) entries, I gave you an alternate solution in my response.

**Comment by sissbruecker** [comment]
No worries, I'm just trying to clarify the problem and see if there is actually a bug with redirects. See the following screencast:

https://user-images.githubusercontent.com/357820/160254364-d92a458e-c09f-43fc-b6f2-39608b113f9e.mp4

What we see here:
- Move to page 2
- Edit a bookmark, add a tag, save
- Redirects back to page 2
- Edit next bookmark, add a tag, save
- Redirects back to page 2
- and so on...

So what I'm trying to figure out is if this somehow / in some cases doesn't work for you? Can this be reproduced somehow?

**Comment by asitemade4u** [comment]
I will try and reproduce it + film.
No matter what the idea of beain able to select all non-tagged records is a good one ;o))

**Comment by asitemade4u** [comment]
I can't reproduce it -- I have the feeling the docker image was very recently updated to the last version because there are plenty of new things I did not know (and I use WatchTower so that I never miss a version).

**Comment by asitemade4u** [comment]
Well, I have found an example that is NOT a bug but will show you how useful it could be:

1. I am tagging entries
2. I have 84 pages of entries (!)
3. I begin at the end because I am sure I will find more un-tagged entries there
4. I am now at page 79 (which does NOT show on the page grid)
5. for some reason I lose the Linkding tab in my browser > I go back to Linkding
6. How am I supposed to find page 79? Yet, that one is relatively simple to access, but what about page 54?

**Comment by KalStormBYU** [comment]
Obviously not an optimal solution, but it is already possible in the URL to add a `page` parameter that will take you directly to the page you want. For example, `https://linkding.mydomain.com/bookmarks?page=54`. This parameter appears as soon as you leave the first page, and can be edited to the page you want.
